File: 1288. Remove Covered Intervals/Solution1288.py
from typing import List


class Solution1288:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        n = len(intervals)
        intervals.sort(key=lambda u:(u[0], -u[1]))
        count, end = 1, intervals[0][1]
        for i in range(1,n):
            if intervals[i][1] > end:
                count += 1
                end = intervals[i][1]
        return count
    

File: 1288. Remove Covered Intervals/Solution1288.java
class Solution1288 { // 03
    public int removeCoveredIntervals(int[][] intervals) {
        Arrays.sort(intervals, 
                    (i1, i2) -> (i1[0] != i2[0]) ? (i1[0] - i2[0]) : (i2[1] - i1[1])
                   );
        int right = intervals[0][1];
        int count = 1;
        int n = intervals.length;
        for (int i = 1; i < n; i++) {
            if (intervals[i][1] > right) {
                count++;
                right = intervals[i][1];
            } 
        }
        return count;
    }
}

// https://leetcode.cn/problems/remove-covered-intervals/solution/by-stormsunshine-dgkn/